hail/python/test/hail/expr/test_ndarrays.py

```diff
@@ -123,24 +125,55 @@ def test_ndarray_shape():
         (m.transpose().shape, np_m.transpose().shape))
 
 @skip_unless_spark_backend()
-@run_with_cxx_compile()
 def test_ndarray_reshape():
+    np_single = np.array([8])
+    single = hl._ndarray([8])
+
     np_a = np.array([1, 2, 3, 4, 5, 6])
     a = hl._ndarray(np_a)
 
     np_cube = np.array([0, 1, 2, 3, 4, 5, 6, 7], order='F').reshape((2, 2, 2))
-    cube = hl._ndarray([0, 1, 2, 3, 4, 5, 6, 7], row_major=False).reshape((2, 2, 2))
+    cube = hl._ndarray([0, 1, 2, 3, 4, 5, 6, 7]).reshape((2, 2, 2))
     cube_to_rect = cube.reshape((2, 4))
     np_cube_to_rect = np_cube.reshape((2, 4))
     cube_t_to_rect = cube.transpose((1, 0, 2)).reshape((2, 4))
     np_cube_t_to_rect = np_cube.transpose((1, 0, 2)).reshape((2, 4))
 
     assert_ndarrays_eq(
+        (single.reshape(()), np_single.reshape(())),
+        (a.reshape((6,)), np_a),
         (a.reshape((2, 3)), np_a.reshape((2, 3))),
         (a.reshape((3, 2)), np_a.reshape((3, 2))),
+        (a.reshape((3, -1)), np_a.reshape((3, -1))),
+        (a.reshape((-1, 2)), np_a.reshape((-1, 2))),
         (cube_to_rect, np_cube_to_rect),
         (cube_t_to_rect, np_cube_t_to_rect))
```

Comment:
Can we add tests like these but with co-prime dimension sizes in the source matrix?

Reply:
And maybe some shape permuting tests? Like `.reshape(3,5,7,9).reshape(9,7,3,5)`